src/modules/commander/commander.cpp

```diff
@@ -2747,6 +2740,20 @@ Commander::run()
 
 		arm_auth_update(now, params_updated || param_init_forced);
 
+		// Handle shutdown request from emergency battery action
+		if(dangerous_battery_level_requests_poweroff){
+			mavlink_log_critical(&mavlink_log_pub, "DANGEROUSLY LOW BATTERY, SHUT SYSTEM DOWN");
+			usleep(2000000);
+			int ret_val = px4_shutdown_request(false, false);
+
+			if (ret_val) {
+				mavlink_log_critical(&mavlink_log_pub, "SYSTEM DOES NOT SUPPORT SHUTDOWN");
```

Comment:
Here we need to reset the flag `dangerous_battery_level_requests_poweroff`

Reply:
Right, thanks :+1: